providers/instantly/delete.go

```diff
@@ -0,0 +1,39 @@
+package instantly
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+)
+
+// Delete removes object. As of now only removal of Campaigns is allowed.
+func (c *Connector) Delete(ctx context.Context, config common.DeleteParams) (*common.DeleteResult, error) {
+	if len(config.ObjectName) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	if len(config.RecordId) == 0 {
+		return nil, common.ErrMissingRecordID
+	}
+
+	if !supportedObjectsByDelete.Has(config.ObjectName) {
+		// Removing campaign is the only to be supported at this time.
```

Comment:
Is this business decision? @laurenzlong? instantly API seems to supprot delete for all objects.

Reply:
no, not a business decision, Constantin can you elaborate on the reason for this?

Reply:
:sweat: the comment line says campaign while it should say `tags`. Thats a typo.

I created a list of supported objects for Read, Write, Delete bound to current design of inpurt params.
Speaking of Delete I created this connector interface to match others. 

I am not sure about importance of usage of Delete. So let me know if I need to spend time for the following upgrade:
* add `common.DeleteResult.Data` field to store user payload
* modify HTTP client's Delete method

Supporting Delete payload will enable the following Instantly operations:
* Delete Account
* Delete Customer Tracking Domain (this is not a List resource)
* Remove entries from blocklist
* Delete leads from a campaign

PS: my main focus was on Read/Write. This is the first connector to challenge simple OOB Delete. Usually it is a good REST API practice to not have payload during Delete. But APIs are different some may do this via POST, etc.

Reply:
@laurenzlong so I think main challenge is that delete does not have single id or key reference that can delete objects. It requires campaign id and and email to delete leads. So basically they use composite key like referencing, so we need to have multiple parameters to delete a lead. I think, if delete is not part of requirement we can skip for now. Otherwise, we need to support multi-primary key parameters. 

@Cobalt0s other parts of code look good, but Lauren should make decision on how we want to support delete. I will delegate PR approval to Lauren.

Reply:
Yes, `multi-primary key parameters` is the general rule that I was using to categorize objects for Read, Write, and by inertia I did it ignore Delete.
Luckily, those operations that I listed above can be managed with single ID or no IDs at all. This information can be passed via `Request Body`, but that would require implementation changes I had mentioned. It depends on the business requirements.

Having delete is nice to roll back create/update which was used as a type of cleanup. People using connectors as a library may find it useful. As of now I don't see usage of `Delete` in server.

Lets wait for the roadmap from Lauren.

Reply:
We don't yet expose the ability to delete to our customers, so only supporting tags in this connector for now is fine - @Cobalt0s can you take a look at my suggestions on the comment line? Hopefully I captured the information correctly.